src/app/Controllers/ParameterValidator.cs

```diff
@@ -179,10 +282,101 @@ public static JsonResult MovieId(string movieId, string method, ILogger logger)
                 !int.TryParse(movieId.Substring(2), out int val) ||
                 val <= 0)
             {
-                return GetAndLogBadParam("Invalid Movie ID parameter", method, logger);
+                return GetAndLogInvalidMovieIdParameter(method, logger);
             }
 
             return null;
         }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="statusCode"></param>
+        /// <param name="errorCode"></param>
+        /// <param name="errorMessage"></param>
+        /// <param name="target"></param>
+        /// <param name="innerErrorType"></param>
+        /// <param name="minValue"></param>
+        /// <param name="maxValue"></param>
+        /// <returns></returns>
+        private static IActionResult BuildJsonResponse(HttpStatusCode statusCode, string errorCode, string errorMessage, string target,
+            InnerErrorType innerErrorType, object minValue, object maxValue)
+        {
+            var sb = new StringBuilder();
+            sb.Append("{\n");
+            sb.Append("\"error\": {\n");
+            sb.Append($"\"code\": \"{errorCode}\",\n");
+            sb.Append($"\"message\": \"{errorMessage}\",\n");
+            sb.Append($"\"statusCode\": {(int)statusCode},\n");
+            sb.Append($"\"target\": \"{target}\",\n");
+            sb.Append("\"innererror\": {\n");
+            sb.Append($"\"code\": \"{innerErrorType}\",\n");
+
+            switch (innerErrorType)
+            {
+                case InnerErrorType.InvalidSearchParameter:
+                    sb.Append($"\"minLength\": {minValue},\n");
+                    sb.Append($"\"maxLength\": {maxValue},\n");
+                    sb.Append("\"characterTypes\": [\n");
+                    sb.Append("\"lowerCase\",\n");
+                    sb.Append("\"upperCase\",\n");
+                    sb.Append("\"number\",\n");
+                    sb.Append("\"symbol\",\n");
+                    sb.Append("]\n");
+                    break;
+                case InnerErrorType.InvalidPageSizeParameter:
+                    sb.Append($"\"minValue\": {minValue},\n");
+                    sb.Append($"\"maxValue\": {maxValue},\n");
+                    sb.Append("\"valueTypes\": [\n");
+                    sb.Append("\"integer\",\n");
+                    sb.Append("]\n");
+                    break;
+                case InnerErrorType.InvalidPageNumberParameter:
+                    sb.Append($"\"minValue\": {minValue},\n");
+                    sb.Append($"\"maxValue\": {maxValue},\n");
+                    sb.Append("\"valueTypes\": [\n");
```

Comment:
for a single item array, i think one line is more readable

valueTypes: [ "integer" ],

comared to
valueTypes:
[
    "integer"
],

Reply:
The reason I split these across lines is that the `InvalidSearchParameter` has multiple value types and adding/removing them is easier to visualize on separate lines. My thought was the same for the other parameters, if you needed to add more, you just add a line.